infra/Pos/Reporting/Methods.hs

```diff
@@ -59,6 +60,13 @@ import           Pos.Reporting.MemState   (MonadReportingMem, rcLoggingConfig,
     lhs' = reverse $ dropWhile isSlash $ reverse lhs
     rhs' = dropWhile isSlash rhs
 
+type ReportingMode m =
```

Comment:
FYI: we decided not to use `Mode` for constraints. Mode is execution mode. `MonadReporting` would be more consistent with the rest of the code.